### PR TITLE
alerts: Update MimirMemoryMapAreasTooHigh to include read-write components

### LINKED ISSUE #3366: Ensure all alerts correctly work when Mimir is deployed using the read-write deployment mode
We should check all Mimir alerts 1-by-1, to ensure all of them correctly work (e.g. metrics are correctly queried) when Mimir is deployed in read-write deployment mode.

For example, any alert querying metrics with the `container` label matcher may not work (because container names are different). We could also consider a linter failing CI if anyone is using `container` label matcher in alerts.

### PR DESCRIPTION
**Pull request by jhesketh**
#### What this PR does

Update MimirMemoryMapAreasTooHigh to include read-write components.

#### Which issue(s) this PR fixes or relates to

Partial fixes #3366

#### Checklist

- [x] Tests updated
- [ ] Documentation added
- [x] `CHANGELOG.md` updated - the order of entries should be `[CHANGE]`, `[FEATURE]`, `[ENHANCEMENT]`, `[BUGFIX]`

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -66,6 +66,7 @@
 * [ENHANCEMENT] Dashboards: improved resources and networking dashboards to work with read-write deployment mode too. #3497 #3504 #3519 #3531
 * [ENHANCEMENT] Alerts: Added "MimirDistributorForwardingErrorRate" alert, which fires on high error rates in the distributor’s forwarding feature. #3200
 * [ENHANCEMENT] Improve phrasing in Overview dashboard. #3488
+* [BUGFIX] Alerts: Fixed `MemoryMapAreasTooHigh` alert when Mimir is deployed in read-write mode. #3626
```
Comment (pracucci) [inline CHANGELOG.md:69]: Rebase `main` and move up.

File: operations/mimir-mixin/config.libsonnet [hunk 2]
```diff
@@ -89,6 +89,9 @@
       write: '(distributor|ingester.*|mimir-write.*)',
       read: '(query-frontend.*|querier.*|ruler-query-frontend.*|ruler-querier.*|mimir-read.*)',
       backend: '(ruler|query-scheduler.*|ruler-query-scheduler.*|store-gateway.*|compactor.*|alertmanager|overrides-exporter|mimir-backend.*)',
+
+      // Specific groups of components.
+      memory_map_areas: '(cortex|ingester.*|store-gateway.*|mimir-write.*|mimir_backend.*)',
```
Comment (pracucci) [inline operations/mimir-mixin/config.libsonnet:94]: Not big fan of this cause adds more additional complexity to the `job_names`. I think you just want to merge `ingester` and `store_gateway`, so you could do the following in the alert (yes, there are some repetitions but it's not a bit deal):

```
$.jobMatcher('(%s|%s)' % [$._config.job_names.ingester, $._config.job_names.store_gateway])
```

Comment (jhesketh) [reply operations/mimir-mixin/config.libsonnet:94]: Fair enough, will do it that wayl

**Review by pracucci — APPROVED** [review]
LGTM, thanks!

**Pull request merged**